pennylane/ops/op_math/adjoint.py

```diff
@@ -401,7 +401,8 @@ def has_generator(self):
         return self.base.has_generator
 
     def generator(self):
-        return qml.s_prod(-1.0, self.base.generator())
+        base_gen = self.base.generator()
+        return qml.Hamiltonian([-1.0 * base_gen.coeffs[0]], base_gen.ops)
```

Comment:
I can see this used to be `-1 * self.base.generator()`, and then the `s_prod` above. What has changed so the we need to separate out the `base_gen.coeffs` now?

Reply:
And what happens here if the `base_gen` is a `Sum`, for example? By unpacking `base_gen.ops`, but still only having a single coefficient, the length of coifs and base_gen won't match, and we'll get an error.